cmd/templatizer/app/renderer/render_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package renderer
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/jaegertracing/jaeger/cmd/templatizer/app"
+	esTemplate "github.com/jaegertracing/jaeger/pkg/es"
+)
+
+func TestIsValidOption(t *testing.T) {
+	tests := []struct {
+		name          string
+		arg           string
+		expectedValue bool
+	}{{name: "span mapping", arg: "jaeger-span", expectedValue: true},
+		{name: "service mapping", arg: "jaeger-service", expectedValue: true},
+		{name: "Invalid mapping", arg: "dependency-service", expectedValue: false},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := IsValidOption(test.arg); got != test.expectedValue {
+				t.Errorf("isValidOption() = %v, want %v", got, test.expectedValue)
+			}
+		})
+	}
+}
+
+func Test_getMappingAsString(t *testing.T) {
```

Comment:
Suggestion to replace this function with below. What's changed:
- Remove fixMappingFunc in favour of mocks.
- Use asserts rather than `if` checks.
- Add line coverage for parse boolean case
```
func Test_getMappingAsString(t *testing.T) {
	tests := []struct {
		name    string
		args    app.Options
		want    string
		wantErr error
	}{
		{
			name: "ES version 7", args: app.Options{Mapping: "jaeger-span", EsVersion: 7, Shards: 5, Replicas: 1, EsPrefix: "test", UseILM: "true"},
			want: "ES version 7",
		},
		{
			name: "ES version 6", args: app.Options{Mapping: "jaeger-span", EsVersion: 6, Shards: 5, Replicas: 1, EsPrefix: "test", UseILM: "false"},
			want: "ES version 6",
		},
		{
			name: "Parse Error version 6", args: app.Options{Mapping: "jaeger-span", EsVersion: 6, Shards: 5, Replicas: 1, EsPrefix: "test", UseILM: "false"},
			wantErr: errors.New("parse error"),
		},
		{
			name: "Parse Error version 7", args: app.Options{Mapping: "jaeger-span", EsVersion: 7, Shards: 5, Replicas: 1, EsPrefix: "test", UseILM: "true"},
			wantErr: errors.New("parse error"),
		},
		{
			name: "Parse bool error", args: app.Options{Mapping: "jaeger-span", EsVersion: 7, Shards: 5, Replicas: 1, EsPrefix: "test", UseILM: "foo"},
			wantErr: errors.New("strconv.ParseBool: parsing \"foo\": invalid syntax"),
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			// Prepare
			mockTemplateApplier := &mocks.TemplateApplier{}
			mockTemplateApplier.On("Execute", mock.Anything, mock.Anything).Return(
				func(wr io.Writer, data interface{}) error {
					wr.Write([]byte(tt.want))
					return nil
				},
			)
			mockTemplateBuilder := &mocks.TemplateBuilder{}
			mockTemplateBuilder.On("Parse", mock.Anything).Return(mockTemplateApplier, tt.wantErr)

			// Test
			got, err := GetMappingAsString(mockTemplateBuilder, &tt.args)

			// Validate
			if tt.wantErr != nil {
				assert.EqualError(t, err, tt.wantErr.Error())
			} else {
				assert.NoError(t, err)
			}
			assert.Equal(t, tt.want, got)
		})
	}
}

```

Reply:
Fixed this in new PR